### DISCUSSION TITLE
Saving Game Data

### DISCUSSION THREAD

**Discussion by soulfir** [pygame-web/pygbag#148] [Q&A] [answered]
Hi,

I'm currently porting a game that was developed in pygame to work on the browser with pygbag and have managed to get the game working nicely. Thanks for the work done on this amazing project!

This game, however, is used as part of a user study and I need to be able to save the in-game data collected during the playthrough. Normally I would use either the email library or the request library to send this information to be stored somewhere, but that doesn't seem to work when running the game on the browser with pygbag. I have hitherto also been unable to find a way to somehow save the data locally so that the study participants could find it and send it to me. 

Is there any way of doing this that is escaping my grasp?

**Comment by pmp-p** [comment, marked as answer]
you could post data https://github.com/pygame-web/pygbag/blob/87f2dcee9df94350d0c0ff87cba5708788acfc44/src/pygbag/support/cross/aio/fetch.py#L193

or download to user a file created by python: `shell.rx("/path/of/file")`

**Reply by soulfir** [reply]
Hi, thanks for the quick reply!

I have tried the posting method you here linked, but I can't get it to work. It works fine running python locally, but when running on the browser with pygbag it always crashes.

As for the download solution you here propose, I'm a bit at a loss as how to implement it. Where should I call shell.rx("/path/of/file")? I can't find any shell library with a rx() method.

**Reply by pmp-p** [reply]
shell is a builtin that handle commands in debug terminal, it always accessible, its code is coming from [cpythonrc.py](https://github.com/pygame-web/pygbag/blob/35607893baf85ca5efd44e0991c4c3cec3765c67/src/pygbag/support/cpythonrc.py#L403) and is readable on CDN as a python file

**Reply by soulfir** [reply]
Thank you so much for the help! I couldn't get it to work directly by calling shell.rx("/path/of/file") as nothing happened, but I did manage to do it by calling platform.window.MM.download("/path/of/file").
